battleship.py

```diff
@@ -163,25 +210,48 @@ def isHorizontal(ship):
 Returns: list of ints
 '''
 def getClickedCell(data, event):
-    return
+    x = int(event.y/data["cell"])
+    y = int(event.x/data["cell"])
+    clickedCell = []
+    clickedCell.append(x)
+    clickedCell.append(y)
+    return clickedCell
 
 
 '''
 drawShip(data, canvas, ship)
 Parameters: dict mapping strs to values ; Tkinter canvas; 2D list of ints
 Returns: None
-'''
+
 def drawShip(data, canvas, ship):
     return
+'''    
 
+def drawShip(data, canvas, ship):
+    cellsize = data["cell"]
+    for row in range(data["rows"]):
+        for col in range(data["cols"]):
+            grid = [row,col]
+            if grid in ship:
+                canvas.create_rectangle(col*cellsize,row*cellsize,(col+1)*cellsize,(row+1)*cellsize,fill="white",width=1)
+            else :
+                canvas.create_rectangle(col*cellsize,row*cellsize,(col+1)*cellsize,(row+1)*cellsize,fill="blue",width=1)  
 
 '''
 shipIsValid(grid, ship)
 Parameters: 2D list of ints ; 2D list of ints
 Returns: bool
 '''
 def shipIsValid(grid, ship):
-    return
+    if (isHorizontal(ship) or isVertical(ship)):
+        if(checkShip(grid,ship) and len(ship) == 3): 
+           return True
+        else :
+            return False    
+    else :

```

Comment:
No need of writing these lines, if the condition fails, it will directly execute the last return.

Reply:
Also, resolve conflicts by deleting pycache and then clicking on the resolve conflicts button for battleship.py